mlflow/models/utils.py

```diff
@@ -863,3 +866,38 @@ def get_model_version_from_model_uri(model_uri):
     (name, version) = get_model_name_and_version(client, model_uri)
     model_version = client.get_model_version(name, version)
     return model_version
+
+
+def _enforce_parameters_schema(params: Optional[Dict[str, Any]], schema: ParamSchema):
```

Comment:
Can we add some tests to validate the functionality of this enforcement with positive and negative examples that covers as many scenarios as possible to guard against future regressions?